tests/suites/test_suite_psa_crypto_driver_wrappers.function

```diff
@@ -1784,3 +1821,195 @@ exit:
     mbedtls_test_driver_hash_hooks = mbedtls_test_driver_hash_hooks_init();
 }
 /* END_CASE */
+
```

Comment:
About testing encryption: one thing that's missing is the behavior of the library when encrypting with a public key. I think this is an unacceptable test gap. Preferably, we should test encryption with a public key even when private key support is disabled (although I think encryption-only builds are rare, unlike verification-only builds).

I propose to add an `asymmetric_encrypt` which does the following:

* Only test with a public key as input (since `asymmetric_encrypt_decrypt` takes care of a private key input).
* Do all the usual testing except for the resulting value.
* For both `RSA_PKCS1V15_CRYPT` and `RSA_OAEP`, check that the output length is the key length.
* In builds with `MBEDTLS_BIGNUM_C`, for both `RSA_PKCS1V15_CRYPT` and `RSA_OAEP`, manually extract the modulus and exponent from the key (or pass them as separate test data parameters), convert the output to an MPI, call `mbedtls_mpi_exp_mod` to obtain the raw plaintext, convert back to a byte string, and make some simple sanity checks on this byte string:
    * [v1.5](https://datatracker.ietf.org/doc/html/rfc8017#section-9.2): the raw plaintext must be `00 01 ff…ff 00 T` where `T` is the input.
    * [OAEP](https://datatracker.ietf.org/doc/html/rfc8017#section-9.1.2): the raw plaintext must start with 0x00 and end with 0xbc.

(Note that this should not use any RSA function, so that it works in builds with a driver and no `MBEDTLS_RSA_C`.)

This way we'll have reasonable confidence that calling an asymmetric encryption driver with a plaintext key actually invokes the driver, returns the expected error, and passes the expected output. We can't validate the _driver_ well that way, but we'll validate the _plumbing_ between the core and the driver, which is the goal of these tests.

Reply:
If I understand correctly, you would just like to duplicate the decryption code in our tests?
If we want to test our plumbing, why cannot we assume that if we do an encryption -> decryption cycle and it works, if someone turns off private key support - the first part will work the same?

Reply:
Can you please post the code you have so far? I'll take a look in a debugger.

Reply:
I pushed some fixes (see changelog message). The test cases now pass in one configuration. There may be remaining minor problems which I haven't checked, e.g. missing test cases or build errors in some configuration.

Reply:
Checked the code and results on my side and the output is now consistent with the assumptions. I understand that the main problems was with:
- modulus: was correct except leading zeros,
- exponent: was invalid as should be used private exponent instead of public one.

I wonder how did you extracted private exponent?

Thank you for your support in this matter.
I will now try to add remaining test driver test cases and fix issues pointed by CI.

Reply:
> I wonder how did you extracted private exponent?

I took a test case with the private key and copied it manually. An RSA private key has: header, modulus, public exponent, private exponent, other numbers in this order. Each number is `0x02`, length, base 256 digits. After a while you get used to parsing ASN.1 manually. Or you can use e.g. http://lapo.it/asn1js/ . Or `openssl rsa -inform DER -text`